cmd/bendo/location_test.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"os"
+	"testing"
+
+	"github.com/ndlib/bendo/store"
+)
+
+const (
+	typeMemory = iota
+	typeFileSystem
+	typeS3
+	typeBlackPearl
+)
+
+func TestSplitBucketPrefix(t *testing.T) {
+	var table = []struct {
+		location string
+		addition string
+		bucket   string
+		prefix   string
+	}{
+		{"", "", "", ""},
+		{"rel/path", "", "rel", "path/"},
+		{"/abs/path/", "", "abs", "path/"},
+		{"/bucket", "", "bucket", ""},
+		{"/bucket", "more", "bucket", "more/"},
+		{"/bucket/prefix/", "", "bucket", "prefix/"},
+		{"/bucket/prefix", "", "bucket", "prefix/"},
+		{"/bucket/prefix", "more", "bucket", "prefix/more/"},
+		{"/bucket/prefix/", "more", "bucket", "prefix/more/"},
+	}
+
+	for _, row := range table {
+		t.Log(row.location, row.addition)
+		bucket, prefix := splitBucketPrefix(row.location, row.addition)
+		if bucket != row.bucket {
+			t.Error("expected bucket", row.bucket, "received", bucket)
+		}
+		if prefix != row.prefix {
+			t.Error("expected prefix", row.prefix, "received", prefix)
+		}
+	}
+}
+
+func TestParseLocation(t *testing.T) {
+	var table = []struct {
+		location string
+		addition string
+		typ      int
+		bucket   string
+		prefix   string
+	}{
+		{"", "", typeMemory, "", ""},
+		{"rel/path", "", typeFileSystem, "", ""},
+		{"/abs/path/", "", typeFileSystem, "", ""},
+		{"file:/rel/path", "", typeFileSystem, "", ""},
+		{"file:rel/path", "", typeFileSystem, "", ""},
+		{"s3:/bucket", "", typeS3, "bucket", ""},
+		{"s3:/bucket", "more", typeS3, "bucket", "more/"},
+		{"s3://localhost:9000/bucket/prefix/", "", typeS3, "bucket", "prefix/"},
+		{"s3://localhost:9000/bucket/prefix/", "more", typeS3, "bucket", "prefix/more/"},
+		{"blackpearl:/bucket", "", typeBlackPearl, "bucket", ""},
+		{"blackpearl:/bucket", "more", typeBlackPearl, "bucket", "more/"},
+		{"blackpearl://192.168.1.70:9000/bucket/prefix/", "", typeBlackPearl, "bucket", "prefix/"},
```

Comment:
No a big deal - why this default value? It's never gonna work, right?

Reply:
This test is for parsing the string. It probably wouldn't work for connecting to it.